### PR TITLE
Remote file management integration

### PR DESCRIPTION
**Pull request by bakatzir**
## Status
Ready
## Related Issues
fixes: https://github.com/demisto/etc/issues/16663
fixes: https://github.com/demisto/etc/issues/13439
fixes: https://github.com/demisto/etc/issues/16227

## Description
create a new integration that handle file on a remote web-server using ssh and scp

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation - https://github.com/demisto/etc/issues/17352
- [x] Code Review

## Dependencies
https://github.com/demisto/dockerfiles/pull/54

### CODE REVIEW

File: Integrations/RemoteFileManagement/RemoteFileManagement.py [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+                                                                                            None) else None
+DOCUMENT_ROOT = '/' + demisto.params().get('document_root') if demisto.params().get('document_root', None) else None
+
+''' UTILS '''
+
+
+def create_certificate_file(password: str, certificate: str):
+    cert_file = tempfile.NamedTemporaryFile()
+    if certificate:
+        with open(cert_file.name, "w") as f:
```
Comment (glicht) [inline Integrations/RemoteFileManagement/RemoteFileManagement.py]: No need to open. NamedTemporaryFile() will return an open file. Also make sure to flush after write.

File: Integrations/RemoteFileManagement/RemoteFileManagement.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+    elif SSH_EXTRA_PARAMS:
+        param_list = ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name] + SSH_EXTRA_PARAMS + [
+            USERNAME + '@' + HOSTNAME, command]
+        result = subprocess.run(param_list, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    else:
+        result = subprocess.run(
+            ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name, USERNAME + '@' + HOSTNAME, command],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+    if result.stderr and result.stderr.find("Warning: Permanently added") == -1:
```
Comment (glicht) [inline Integrations/RemoteFileManagement/RemoteFileManagement.py]: We should check also result.returncode. Stderr can be empty and still fail.

Comment (bakatzir) [reply Integrations/RemoteFileManagement/RemoteFileManagement.py]: @glicht please see my related fixes

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 3]
```diff
@@ -0,0 +1,537 @@
...
+            demisto.results('ok')
+
+        elif demisto.command() == 'pan-os-edl-get-external-file':
+            rfm_get_external_file_command()
+
+        elif demisto.command() == 'pan-os-edl-search-external-file':
+            rfm_search_external_file_command()
+
+        elif demisto.command() == 'pan-os-edl-update':
+            rfm_update()
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: change to edl-updat_command()

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 4]
```diff
@@ -0,0 +1,537 @@
...
+''' EXECUTION '''
+
+
+def main():
+    LOG('command is %s' % (demisto.command(),))
+    try:
+        if demisto.command() == 'test-module':
+            ssh_execute('echo 1')
+            demisto.results('ok')
+
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: change all function names from `rfm_X_command` to `edl_x_command`

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 5]
```diff
@@ -0,0 +1,537 @@
...
+        else:
+            return_error('Command failed with exit status: ' + str(result.returncode))
+    else:
+        return True
+
+
+''' COMMANDS '''
+
+
+def rfm_get_external_file(file_path: str):
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: change name to edl_get_external_file

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+
+    md = tableToMarkdown('File Content:', result, headers=['List'])
+    demisto.results({
+        'ContentsFormat': formats['markdown'],
+        'Type': entryTypes['note'],
+        'Contents': md
+    })
+
+
+def rfm_search_external_file(file_path: str, search_string: str):
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: change name to edl_search_external_file

File: Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+        })
+    else:
+        demisto.results({
+            'Type': 11,
+            'Contents': 'Search string was not found in the external file path given.',
+            'ContentsFormat': formats['text']
+        })
+
+
+def rfm_update_external_file(file_path: str, list_name: str, verbose: bool) -> bool:
```
Comment (yuvalbenshalom) [inline Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.py]: change name to edl_update_external_file

**Comment by bakatzir** [comment]
@kirbles19 , not to many outputs, but would ask you to review detailed description as well:
https://github.com/demisto/content/blob/dd08fbfbcc6d6f3728c661ec8cf901281cf3f5d5/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.yml

https://github.com/demisto/content/blob/dd08fbfbcc6d6f3728c661ec8cf901281cf3f5d5/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md

File: Integrations/RemoteFileManagement/RemoteFileManagement.py (hunk 2 above)
Comment (glicht) [reply Integrations/RemoteFileManagement/RemoteFileManagement.py]: Looks good

**Comment by glicht** [comment]
@kirbles19 links to files you can edit directly:

https://github.com/demisto/content/blob/remote_file_managment_integration/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management.yml


https://github.com/demisto/content/blob/remote_file_managment_integration/Integrations/PaloAltoNetworks_PAN_OS_EDL_Management/PaloAltoNetworks_PAN_OS_EDL_Management_description.md

**Comment by kirbles19** [comment]
@bakatzir All done with .yml, .md, and error messages in .py.

**Pull request merged**